File: KP 2020 Java/src/l3/Quadrangle.java
package l3;

/**
 * @author Tobiasz Wojnar
 * Java Programing Course 2020
 * list 3 exercise 1
 * Calculate surface area and circumference of:
 * circle, quadrangle (square, rectangle, rhombus), regular pentagon, regular hexagon
 * implement hierarchy and abstract class Figure
 * handle exeptions and bad data
 *
 * @input first letter of figure and paramethers
 *	for circle			- radious
 *	for quadrangle			- 4 sides and angle
 *	for pentagon and hexagon	- side
 * eg.: java figures q 8 8 4 4 90
 **/

public abstract class Quadrangle extends Figure {
    double[] sides = new double[4];
    double angle;
    public Quadrangle (double side1, double side2, double side3, double side4, double angle){
        sides[0]=side1;
        sides[1]=side2;
        sides[2]=side3;
        sides[3]=side4;
        this.angle=angle;
    }
    public double Circumference () {
        return sides[0]+sides[1]+sides[2]+sides[3];
    }
}

File: kp 2020 cpp/l3/Quadrangle.cpp
/**
 * Creates quadrangle
 * It extends abstract class Figure and inheritance method for calculating circumference
 * It is extended by 3 subclasses Rhombus, Rectangle and Square
 *
 * @author Tobiasz Wojnar
 * excise for university course
 * @see Main.cpp for full requriments
 * @see Rhombus.hpp
 * @see Rectangle.hpp
 * @see Square.hpp
 * @see Figure.hpp
 */

#include <iostream>
#include "Quadrangle.hpp"
/**
 * Constructor of quadrangle, since we create only regular hexagons it needs only one parameter
 * @param side1, side2, side3, side4 - sides of quadrangle, are positive, double precision numbers
 * @param angle in degrees;  double precision number, in range (0,180)
 */
Quadrangle :: Quadrangle (double side1,double side2,double side3,double side4, double angle){
    this->side1=side1;
    this->side2=side2;
    this->side3=side3;
    this->side4=side4;
    this->angle=angle;
}
/**
 * @return circumference of quadrangle, sum of its sides
 */
double Quadrangle :: Circumference () {
    return side1+side2+side3+side4;
}